pkg/assembler/backends/testing/pkg.go

```diff
@@ -409,6 +427,47 @@ func (c *demoClient) buildPackageResponse(id uint32, filter *model.PkgSpec) (*mo
 	return &p, nil
 }
 
+func getPackageIDFromInput(c *demoClient, input model.PkgInputSpec, pkgMatchType model.MatchFlags) (*uint32, error) {
```

Comment:
Does the return have to be pointer to ID?

Reply:
in order to return a nil value returning a pointer.